### ISSUE TITLE
Unable to install version 0.37

### ISSUE THREAD

**Issue by MikeyBurkman** [mongo-express/mongo-express#316]
## This is now solved. Please use v0.37.2 or more!
---

Trying to install version 0.37 gives me webpack linting errors during the build. This happens with both Node 4.4.3 (running npm 2.15.1) and 6.2.0 (npm 3.8.9). I'm running OSX El Capitan.

```
ERROR in collection-5dd057f6d6b8be808685.min.js from UglifyJs
SyntaxError: Unexpected character '`' [collection-5dd057f6d6b8be808685.min.js:45,9]

ERROR in index-d7b21537994dd0a9710b.min.js from UglifyJs
SyntaxError: Unexpected token: punc ()) [index-d7b21537994dd0a9710b.min.js:12,64]

ERROR in gridfs-29ce829200ee5010f2fc.min.js from UglifyJs
SyntaxError: Unexpected character '`' [gridfs-29ce829200ee5010f2fc.min.js:8,18]

ERROR in document-d5ae257b82aeb0f2dd52.min.js from UglifyJs
SyntaxError: Unexpected character '`' [document-d5ae257b82aeb0f2dd52.min.js:50,9]

ERROR in database-d1a46de72525f31e2804.min.js from UglifyJs
SyntaxError: Unexpected token: punc ()) [database-d1a46de72525f31e2804.min.js:12,64]
```

I'm also seeing errors trying to copy css files when running in Node 6 with npm 3. These errors do not show up using npm 2.x, which makes sense, since the directory structure with npm 2 is deterministic, where the dependencies may (and do) get flattened with npm 3.
```
ERROR in [copy-webpack-plugin] unable to locate 'node_modules/codemirror/lib/codemirror.css' at '/Users/myuser/myproject/node_modules/mongo-express/node_modules/codemirror/lib/codemirror.css'

ERROR in [copy-webpack-plugin] unable to locate 'node_modules/bootstrap/dist/css/bootstrap.min.css' at '/Users/myuser/myproject/node_modules/mongo-express/node_modules/bootstrap/dist/css/bootstrap.min.css'

ERROR in [copy-webpack-plugin] unable to locate 'node_modules/bootstrap/dist/css/bootstrap.min.css.map' at '/Users/myuser/myproject/node_modules/mongo-express/node_modules/bootstrap/dist/css/bootstrap.min.css.map'

ERROR in [copy-webpack-plugin] unable to locate 'node_modules/bootstrap/dist/css/bootstrap-theme.min.css' at '/Users/myuser/myproject/node_modules/mongo-express/node_modules/bootstrap/dist/css/bootstrap-theme.min.css'

ERROR in [copy-webpack-plugin] unable to locate 'node_modules/bootstrap/dist/css/bootstrap-theme.min.css.map' at '/Users/myuser/myproject/node_modules/mongo-express/node_modules/bootstrap/dist/css/bootstrap-theme.min.css.map'
```

**Comment by dozoisch** [comment]
@MikeyBurkman Thanks, I'll retest tonight. 

Seems like an issue with the babel config or something that wasn't picked up on my machine for some reason. Seems like the generated code is not in es5 as it should have been. (Weird that travis didn't pick it up because we do run build on travis.)

About the structure... I'll try to change it to use npm to resolve the path. It's curious though because normally these package should've been installed properly (especially with npm 3). Can you confirm whether or not the packages are properly installed (bootstrap and codemirror)?

**Comment by dozoisch** [comment]
Can you also give me the exact step you did to install?

**Comment by MikeyBurkman** [comment]
Pretty much any way you choose to install, the build fails. I even tried in a completely blank project (just did an npm init on an empty directory). `npm i--save mongo-express@0.37.0` fails in Node 6/npm 3. It downloads all the dependencies into a flat structure in node_modules, 452 directories in all. (Ouch.) 

I'd love to tell you the output of the files that is failing linting, but npm deletes mongo-express/ after the build fails. Is there any reason to have clients of mongo-express do this build? Could we instead do the build as part of a pre-publish step, and only include the dist files in npm? 

In a somewhat related issue, I had my friend try to build this on Windows to see if he had the same issue, but the build script fails because env vars need to be set differently on Windows. I suggest using [cross-env](https://www.npmjs.com/package/cross-env) to set NODE_ENV.

**Comment by MikeyBurkman** [comment]
Also, of interest -- I can build mongo-express locally using `cd mongo-express && npm i` with no issue. It's only when I go to include it in another project that it fails to install.

**Comment by dozoisch** [comment]
@MikeyBurkman yeah, thats the used case I couldn't really test before publishing unfortunately. I always tested when installing myself. I'll fix it for usage as a dep. I'll probably end up packaging pre-built scripts

**Comment by MikeyBurkman** [comment]
Yeah that's the trickiest solution, but probably the best one in the long run. Let me know if you need help from me at all. My client is liking this project, so I can probably squeeze in some (billable) time to work on it if it's nothing too involved.

**Comment by dozoisch** [comment]
@MikeyBurkman well, if you are able to change the build step to be a pre-publish that would solve most of the issues I think!

**Comment by MikeyBurkman** [comment]
Of course you pick the one thing I've never actually done :) I think it's as simple as changing the `postinstall` script to `prepublish` and fixing up .npmignore to include the build directory, but I'll play around with it it a bit, probably tomorrow morning.

**Comment by dozoisch** [comment]
(whoops) Perfect. I might have some time tonight so we will see

**Comment by cesarve77** [comment]
same issue:

sudo npm i -g mongo-express
/usr/local/bin/mongo-express -> /usr/local/lib/node_modules/mongo-express/app.js

> fsevents@1.1.1 install /usr/local/lib/node_modules/mongo-express/node_modules/fsevents
> node install

[fsevents] Success: "/usr/local/lib/node_modules/mongo-express/node_modules/fsevents/lib/binding/Release/node-v48-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile

> mongo-express@0.37.0 postinstall /usr/local/lib/node_modules/mongo-express
> npm run build


> mongo-express@0.37.0 build /usr/local/lib/node_modules/mongo-express
> NODE_ENV=production webpack

clean-webpack-plugin: /usr/local/lib/node_modules/mongo-express/build has been removed.
Hash: aa220e46eb368d131399
Version: webpack 2.2.1
Time: 5074ms
                                   Asset       Size  Chunks                    Chunk Names
               css/theme/lesser-dark.css     2.6 kB          [emitted]         
  codemirror-db5b45bbb85d6b6df453.min.js     174 kB       0  [emitted]         codemirror
       index-d7b21537994dd0a9710b.min.js    2.01 kB       2  [emitted]         index
      gridfs-29ce829200ee5010f2fc.min.js  250 bytes       3  [emitted]         gridfs
    document-d5ae257b82aeb0f2dd52.min.js    2.32 kB       4  [emitted]         document
    database-d1a46de72525f31e2804.min.js    2.06 kB       5  [emitted]         database
      vendor-900073eca980643c640e.min.js     126 kB       6  [emitted]         vendor
                         img/favicon.ico    17.8 kB          [emitted]         
              img/mongo-express-logo.png    17.8 kB          [emitted]         
                           img/gears.gif    50.3 kB          [emitted]         
              img/mongo-express-logo.svg    1.44 kB          [emitted]         
               img/mongo-express-logo.ai     386 kB          [emitted]  [big]  
                           css/style.css    1.84 kB          [emitted]         
                      css/codemirror.css    8.13 kB          [emitted]         
                  css/theme/3024-day.css    1.99 kB          [emitted]         
                css/theme/3024-night.css    2.08 kB          [emitted]         
                    css/theme/abcdef.css    1.94 kB          [emitted]         
           css/theme/ambiance-mobile.css  103 bytes          [emitted]         
                  css/theme/ambiance.css    26.5 kB          [emitted]         
                    css/theme/bespin.css    1.41 kB          [emitted]         
               css/theme/base16-dark.css    2.11 kB          [emitted]         
              css/theme/base16-light.css    2.11 kB          [emitted]         
                    css/theme/cobalt.css     1.7 kB          [emitted]         
                css/theme/blackboard.css    1.93 kB          [emitted]         
                css/theme/colorforth.css    1.65 kB          [emitted]         
                   css/theme/dracula.css    2.01 kB          [emitted]         
              css/theme/duotone-dark.css    2.58 kB          [emitted]         
             css/theme/duotone-light.css    2.69 kB          [emitted]         
                   css/theme/eclipse.css    1.16 kB          [emitted]         
               css/theme/erlang-dark.css    2.25 kB          [emitted]         
                   css/theme/isotope.css    1.44 kB          [emitted]         
                 css/theme/hopscotch.css    1.47 kB          [emitted]         
                  css/theme/icecoder.css     2.5 kB          [emitted]         
                 css/theme/liquibyte.css    3.93 kB          [emitted]         
                   css/theme/elegant.css  781 bytes          [emitted]         
  collection-5dd057f6d6b8be808685.min.js    19.4 kB       1  [emitted]         collection
                  css/theme/material.css    2.44 kB          [emitted]         
                       css/theme/mbo.css    2.11 kB          [emitted]         
                   css/theme/monokai.css    1.92 kB          [emitted]         
                  css/theme/midnight.css    2.14 kB          [emitted]         
                      css/theme/neat.css  688 bytes          [emitted]         
                  css/theme/mdn-like.css    5.17 kB          [emitted]         
              css/theme/panda-syntax.css    1.77 kB          [emitted]         
              css/theme/paraiso-dark.css    2.08 kB          [emitted]         
                       css/theme/neo.css  947 bytes          [emitted]         
                     css/theme/night.css    1.72 kB          [emitted]         
             css/theme/paraiso-light.css    2.08 kB          [emitted]         
            css/theme/pastel-on-dark.css    2.45 kB          [emitted]         
                css/theme/railscasts.css    1.51 kB          [emitted]         
                  css/theme/rubyblue.css    1.77 kB          [emitted]         
                      css/theme/seti.css    1.98 kB          [emitted]         
                css/theme/the-matrix.css    1.91 kB          [emitted]         
                 css/theme/solarized.css    5.44 kB          [emitted]         
   css/theme/tomorrow-night-eighties.css    2.44 kB          [emitted]         
                      css/theme/ttcn.css    2.42 kB          [emitted]         
     css/theme/tomorrow-night-bright.css    1.77 kB          [emitted]         
                  css/theme/twilight.css    2.13 kB          [emitted]         
               css/theme/vibrant-ink.css    2.11 kB          [emitted]         
                   css/theme/xq-dark.css       3 kB          [emitted]         
                  css/theme/xq-light.css    2.23 kB          [emitted]         
                      css/theme/yeti.css    1.86 kB          [emitted]         
                   css/theme/zenburn.css    1.99 kB          [emitted]         
  fonts/glyphicons-halflings-regular.eot    20.1 kB          [emitted]         
 fonts/glyphicons-halflings-regular.woff    23.4 kB          [emitted]         
  fonts/glyphicons-halflings-regular.svg     109 kB          [emitted]         
  fonts/glyphicons-halflings-regular.ttf    45.4 kB          [emitted]         
fonts/glyphicons-halflings-regular.woff2      18 kB          [emitted]         
                   css/bootstrap.min.css     121 kB          [emitted]         
               css/bootstrap.min.css.map     542 kB          [emitted]         
             css/bootstrap-theme.min.css    23.4 kB          [emitted]         
         css/bootstrap-theme.min.css.map    25.6 kB          [emitted]         
   [0] ./~/jquery/dist/jquery.js 267 kB {6} [built]
   [1] ./lib/scripts/codeMirrorLoader.js 113 bytes {0} [built]
   [2] ./~/codemirror/lib/codemirror.js 352 kB {0} [built]
   [3] ./lib/scripts/jQueryLoader.js 55 bytes {6} [built]
   [4] ./~/bootstrap/dist/js/npm.js 484 bytes {6} [built]
   [5] ./~/codemirror/mode/javascript/javascript.js 30.9 kB {0} [built]
   [6] ./~/renderjson/renderjson.js 9.6 kB {1} [built]
  [11] ./~/bootstrap/js/collapse.js 5.99 kB {6} [built]
  [13] ./~/bootstrap/js/modal.js 9.99 kB {6} [built]
  [19] ./lib/scripts/collection.js 6.95 kB {1} [built]
  [20] ./lib/scripts/database.js 1.2 kB {5} [built]
  [21] ./lib/scripts/document.js 1.4 kB {4} [built]
  [22] ./lib/scripts/gridfs.js 164 bytes {3} [built]
  [23] ./lib/scripts/index.js 1.16 kB {2} [built]
  [24] ./lib/scripts/vendor.js 143 bytes {6} [built]
    + 10 hidden modules

ERROR in collection-5dd057f6d6b8be808685.min.js from UglifyJs
SyntaxError: Unexpected character '`' [collection-5dd057f6d6b8be808685.min.js:45,9]

ERROR in index-d7b21537994dd0a9710b.min.js from UglifyJs
SyntaxError: Unexpected token: punc ()) [index-d7b21537994dd0a9710b.min.js:12,64]

ERROR in gridfs-29ce829200ee5010f2fc.min.js from UglifyJs
SyntaxError: Unexpected character '`' [gridfs-29ce829200ee5010f2fc.min.js:8,18]

ERROR in document-d5ae257b82aeb0f2dd52.min.js from UglifyJs
SyntaxError: Unexpected character '`' [document-d5ae257b82aeb0f2dd52.min.js:50,9]

ERROR in database-d1a46de72525f31e2804.min.js from UglifyJs
SyntaxError: Unexpected token: punc ()) [database-d1a46de72525f31e2804.min.js:12,64]

npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! mongo-express@0.37.0 build: `NODE_ENV=production webpack`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the mongo-express@0.37.0 build script 'NODE_ENV=production webpack'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mongo-express package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     NODE_ENV=production webpack
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mongo-express
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mongo-express
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /usr/local/lib/node_modules/mongo-express/npm-debug.log
/usr/local/lib
└── (empty)

npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i" "-g" "mongo-express"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! mongo-express@0.37.0 postinstall: `npm run build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mongo-express@0.37.0 postinstall script 'npm run build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mongo-express package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mongo-express
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mongo-express
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/cesar/Documents/www/choice-meteor/npm-debug.log
npm ERR! code 1

**Comment by dozoisch** [comment]
@cesarve77 yup that's due to it being built inside a node module. We need to change build to be a pre-publish step. We do accept help if you have time to tackle this issue right now. Else you can pin to 0.35 right now. This is on my priority list when I have some time off work.

**Comment by dozoisch** [comment]
Fixed as of v0.37.2
